src/Bicep.LangServer/Configuration/BicepConfigChangeHandler.cs

```diff
@@ -0,0 +1,84 @@
+{
+    public class BicepConfigChangeHandler : IBicepConfigChangeHandler
+    {
+        private JSchema bicepConfigSchema;
+        private readonly IFileResolver fileResolver;
+
+        public BicepConfigChangeHandler(IFileResolver fileResolver)
+        {
+            this.fileResolver = fileResolver;
+            var assembly = Assembly.GetExecutingAssembly();
+            string manifestResourceName = assembly.GetManifestResourceNames().Where(p => p.EndsWith("bicepconfig.schema.json", StringComparison.Ordinal)).First();
+            Stream? stream = assembly.GetManifestResourceStream(manifestResourceName);
+            var streamReader = new StreamReader(stream ?? throw new ArgumentNullException("Stream is null"), Encoding.Default);
+
+            bicepConfigSchema = JSchema.Parse(streamReader.ReadToEnd());
+        }
+
+        public void RetriggerCompilationOfSourceFilesInWorkspace(ICompilationManager compilationManager, FileEvent bicepConfigFileEvent, IWorkspace workspace)
+        {
+            Uri bicepConfigUri = bicepConfigFileEvent.Uri.ToUri();
+
+            if (ShouldUpsertCompilation(bicepConfigFileEvent.Type, bicepConfigUri))
+            {
+                foreach (ISourceFile sourceFile in workspace.GetSourceFilesForDirectory(bicepConfigUri))
+                {
+                    Uri uri = sourceFile.FileUri;
+
+                    if (fileResolver.TryRead(uri, out string? bicepFileContents, out ErrorBuilderDelegate _) &&
+                        !string.IsNullOrWhiteSpace(bicepFileContents))
+                    {
+                        compilationManager.UpsertCompilation(DocumentUri.From(uri), null, bicepFileContents);
+                    }
+                }
+            }
+        }
+
+        private bool ShouldUpsertCompilation(FileChangeType fileChangeType, Uri bicepConfigUri)
+        {
+            // When fileChangeType is "Changed", we don't want to retrigger compilation for every key stroke.
+            // We'll check the validity of file against the schema and retrigger compilation only if the file is valid
+            if (fileChangeType == FileChangeType.Changed)
+            {
+                return fileResolver.TryRead(bicepConfigUri, out string? bicepConfigFileContents, out ErrorBuilderDelegate _) &&
+                    bicepConfigFileContents is not null &&
+                    IsBicepConfigValid(bicepConfigFileContents);
+            }
+
+            return true;
+        }
+
+        private bool IsBicepConfigValid(string bicepConfig)
+        {
+            try
+            {
+                JObject jObject = JObject.Parse(bicepConfig);
+                return jObject.IsValid(bicepConfigSchema);
+            }
+            catch
+            {
+                // We want to push error dignostics in case of invalid bicepconfig.json.
+                // We won't be able to apply validation until user fixes issues in config file
+                return true;
```

Comment:
Why return `true` if it's unparseable?

Reply:
This will enable to re trigger compilation and push error diagnostics. RetriggerCompilationOfSourceFilesInWorkspace_WithInvalidBicepConfigFile_ShouldRetriggerCompilation unit test covers this scenario. Let me know if I am missing something here

Reply:
This code doesn't exist anymore. Made updates here:
https://github.com/Azure/bicep/commit/7c53107f3fb09a3e6403cf76b8ee6a876c13806c